Sources/MapboxMaps/Foundation/Camera/FlyToCameraAnimator.swift

```diff
@@ -0,0 +1,117 @@
+import UIKit
+
+public class FlyToCameraAnimator: NSObject, CameraAnimator, CameraAnimatorInterface {
+
+    internal private(set) weak var delegate: CameraAnimatorDelegate?
+
+    public private(set) var owner: AnimationOwner
+
+    internal private(set) var flyToInterpolator: FlyToInterpolator?
+
+    internal private(set) var animationDuration: TimeInterval?
+
+    public private(set) var state: UIViewAnimatingState = .inactive
+
+    internal private(set) var startTime: Date?
+
+    internal private(set) var endTime: Date?
+
+    internal private(set) var finalCameraOptions: CameraOptions?
+
+    internal private(set) var animationCompletion: AnimationCompletion?
+
+    internal init(delegate: CameraAnimatorDelegate,
+                  owner: AnimationOwner = .custom(id: "fly-to")) {
+
+        self.delegate = delegate
+        self.owner = owner
+    }
+
+    deinit {
+        scheduleCompletionIfNecessary(position: .current)
+    }
+
+    internal func makeFlyToInterpolator(from initalCamera: CameraOptions, to finalCamera: CameraOptions, duration: TimeInterval? = nil, screenFullSize: CGSize) {
```

Comment:
we can combine this with the init method and make more of the properties `let`s and non-optional. I tried this out and found that it eliminates some edge cases that have to be handled with the current design.

Reply:
yep it was something I was planning on doing as well as an iteration. But maybe we should pull it forward.

Reply:
refactored on my branch